apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPool.TargetedSessionPool.cs

```diff
@@ -90,24 +90,27 @@ internal TargetedSessionPool(SessionPool parent, DatabaseName databaseName, bool
                 _createSessionRequest = new CreateSessionRequest { DatabaseAsDatabaseName = databaseName };
                 if (acquireSessionsImmediately)
                 {
-                    StartAcquisitionTasksIfNecessary();
+                    StartAcquisitionTasksIfNecessary(0);
                 }
             }
 
             public async Task<PooledSession> AcquireSessionAsync(TransactionOptions transactionOptions, CancellationToken cancellationToken)
             {
-                // We may need to start more tasks just acquire a session. We need to try this *before* counting the requested
-                // session as being active. (If we have 99 active sessions at the moment, and nothing in the pool, we want to start a task
-                // to create the session we'll end up acquiring. If we wait until we've counted this session as being active, we won't
-                // start the new task as it'll take us over the limit.)
-                // FIXME: We don't cope with the "Minimum pool size of 0, max active = 100, currently active = 99" case.
-                StartAcquisitionTasksIfNecessary();
-
                 bool success = false;
                 try
                 {
-                    Interlocked.Increment(ref _activeSessionCount);
-                    PooledSession session = await AcquireSessionImplAsync(transactionOptions, cancellationToken).ConfigureAwait(false);
+                    int sessionCount = Interlocked.Increment(ref _activeSessionCount);
+                    if (sessionCount > Options.MaximumActiveSessions && Options.WaitOnResourcesExhausted == ResourcesExhaustedBehavior.Fail)
```

Comment:
This is the previous active session count, so I think this needs to be `>=`

Reply:
Nope, Interlocked.Increment returns the new count. Do you think it's worth a reminder of that as a comment?